test/poll_for_state.go

```diff
@@ -109,9 +117,43 @@ func (p blockPoller) findTimeout(ctx context.Context, height uint64, packet ibc.
 		return zero, err
 	}
 	for _, t := range timeouts {
+		p.pollErr.PushSearched(t)
 		if t.Packet.Equal(packet) {
 			return t, nil
 		}
 	}
 	return zero, ErrNotFound
 }
+
+type pollError struct {
+	error
+	targetPacket    ibc.Packet
+	searchedPackets []string
+}
+
+func (pe *pollError) SetErr(err error) {
+	pe.error = err
+}
+
+func (pe *pollError) PushSearched(packet interface{}) {
+	pe.searchedPackets = append(pe.searchedPackets, spew.Sdump(packet))
+}
+
+func (pe *pollError) Unwrap() error {
+	return pe.error
+}
+
+func (pe *pollError) Format(s fmt.State, verb rune) {
+	if verb != 'v' && !s.Flag('+') {
+		fmt.Fprint(s, pe.error.Error())
+		return
+	}
+
+	searched := strings.Join(pe.searchedPackets, "\n")
```

Comment:
Technically this does not match the documented purpose of `%+v`:

```
%v	the value in a default format
	when printing structs, the plus flag (%+v) adds field names
```

but given the limited scope we are using it, this is fine. A comment mentioning the intended use with `require` wouldn't hurt here.

Reply:
I stole this pattern from `pkg/errors` (now archived which makes me sad). In that package, it would print the stack trace.

Good point on `require`, will add.